gnomad_qc/v5/sample_qc/generate_qc_mt.py

```diff
@@ -0,0 +1,274 @@
+            mt = mt._filter_partitions(range(2))
+
+        # Filter to gnomAD QC sites and use `NO_HQ_GENOTYPES` to filter to `adj`.
+        mt = mt.semi_join_rows(ht)
+        mt = mt.filter_rows(~mt.filters.contains("NO_HQ_GENOTYPES"))
+        mt = mt.select_rows()
+
+        # Remove samples to exclude and their variant sites.
+        mt = mt.filter_cols(~s_to_exclude.contains(mt.s))
+        mt = mt.filter_rows(hl.agg.any(mt.GT.is_non_ref()))
+        return mt
+
+    logger.info("Filtering AoU ACAF and exome MatrixTables and unioning (on rows)...")
+    # NOTE: Add checkpoints after each MT if run into errors in AoU workbench.
+    acaf_mt = _filter_aou_mt(acaf_mt)
+    exome_mt = _filter_aou_mt(exome_mt)
+    return acaf_mt.union_rows(exome_mt)
+
+
+def generate_qc_mt(
+    gnomad_mt: hl.MatrixTable,
+    aou_mt: hl.MatrixTable,
+    n_partitions: int = 5000,
+) -> hl.MatrixTable:
+    """
+    Union gnomAD v4 QC MT and AoU v8 MTs.
+
+    :param gnomad_mt: gnomAD v4 QC MatrixTable.
+    :param aou_mt: Joint AoU (ACAF + exome) v8 MatrixTable.
+    :param n_partitions: Number of desired partitions for the unioned MatrixTable.
+        Default is 5000.
+    :return: Joint gnomAD v4 (exomes + genomes) + AoU v8 (genomes) QC MT.
+    """
+    logger.info("Resolving sample ID collisions...")
+    sample_collisions = sample_id_collisions.ht()
+    gnomad_mt = add_project_prefix_to_sample_collisions(
+        t=gnomad_mt,
+        sample_collisions=sample_collisions,
+        project="gnomad",
+    )
+    aou_mt = add_project_prefix_to_sample_collisions(
+        t=aou_mt,
+        sample_collisions=sample_collisions,
+        project="aou",
+    )
+
+    joint_mt = gnomad_mt.union_cols(aou_mt)
+    return joint_mt.naive_coalesce(n_partitions)
+
+
+def main(args):
+    """Create a joint gnomAD + AoU dense MT of a diverse set of variants for relatedness/genetic ancestry PCA."""
+    hl.init(
+        log="/home/jupyter/workspaces/gnomadproduction/generate_qc_mt.log",
+        tmp_dir=f"gs://{WORKSPACE_BUCKET}/tmp/4_day",
+    )
+    hl.default_reference("GRCh38")
+
+    overwrite = args.overwrite
+    test = args.test
+
+    try:
+        if args.union_aou_mts:
+
+            logger.info(
+                "Loading AoU ACAF and exome MatrixTables and removing unnecessary entry annotations..."
+            )
+            entry_annotations = args.entry_annotations.split(",")
+            acaf_mt = aou_acaf_mt.mt().select_entries(*entry_annotations)
+            exome_mt = aou_exome_mt.mt().select_entries(*entry_annotations)
+
+            logger.info("Generating joint AoU MatrixTable...")
+            mt = union_aou_mts(
+                acaf_mt=acaf_mt,
+                exome_mt=exome_mt,
+                s_to_exclude=get_samples_to_exclude(
+                    filter_samples=hard_filtered_samples.ht()
+                ),
+                ht=v4_sample_qc.get_joint_qc().mt().rows(),
```

Comment:
`ht=v4_sample_qc.get_joint_qc().mt().rows()` is used a couple times, maybe just define it above